app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/registration/SubscriptionRecoveryController.scala

```diff
@@ -137,7 +137,7 @@ class SubscriptionRecoveryController @Inject() (
           getDateOfBirthOrDateOfEstablishment(
             subscriptionDisplayResponse,
             subscriptionDetails.dateEstablished,
-            subscriptionDetails.dateOfBirth
+            subscriptionDetails.dateOfBirth orElse subscriptionDetails.nameDobDetails.map(_.dateOfBirth)
```

Comment:
I would be tempted to move that logic into the subscriptionDetails object but with a method name to indicate that it's only for recovery.